src/relations.jl

```diff
@@ -2766,39 +2833,162 @@ given a phase partition, `q`.
+    Rv_over_Rd = TP.Rv_over_Rd(param_set)
     q_vap = vapor_specific_humidity(q)
-    return molmass_ratio * shum_to_mixing_ratio(q_vap, q.tot)
+    return Rv_over_Rd * shum_to_mixing_ratio(q_vap, q.tot)
 end
 vol_vapor_mixing_ratio(param_set, ts::ThermodynamicState) =
     vol_vapor_mixing_ratio(param_set, PhasePartition(param_set, ts))
 
+"""
+    partial_pressure_dry(param_set, p, q)
+
+The partial pressure of dry air, given
+
+ - `param_set` an `AbstractParameterSet`, see the [`Thermodynamics`](@ref) for more details
+ - `p` air pressure
+ - `q` phase partition
+"""
+@inline function partial_pressure_dry(
+    param_set::APS,
+    p::FT,
+    q::PhasePartition{FT},
+) where {FT <: Real}
+    Rv_over_Rd = TP.Rv_over_Rd(param_set)
+    return p * (1 - q.tot) /
+           (1 - q.tot + vapor_specific_humidity(q) * Rv_over_Rd)
+end
+
+"""
+    partial_pressure_dry(param_set, ts::ThermodynamicState)
+
+The partial pressure of dry air, given a thermodynamic state `ts`.
+"""
+@inline partial_pressure_dry(
+    param_set::APS,
+    ts::ThermodynamicState{FT},
+) where {FT <: Real} = partial_pressure_dry(
+    param_set,
+    air_pressure(param_set, ts),
+    PhasePartition(param_set, ts),
+)
+
+@inline partial_pressure_dry(
+    param_set::APS,
+    ts::AbstractPhaseDry{FT},
+) where {FT <: Real} = air_pressure(param_set, ts)
+
+"""
+    partial_pressure_vapor(param_set, p, q)
+
+The partial pressure of water vapor, given
+
+ - `param_set` an `AbstractParameterSet`, see the [`Thermodynamics`](@ref) for more details
+ - `p` air pressure
+ - `q` phase partition
+"""
+@inline function partial_pressure_vapor(
+    param_set::APS,
+    p::FT,
+    q::PhasePartition{FT},
+) where {FT <: Real}
+    Rv_over_Rd = TP.Rv_over_Rd(param_set)
+    return p * vapor_specific_humidity(q) * Rv_over_Rd /
+           (1 - q.tot + vapor_specific_humidity(q) * Rv_over_Rd)
+end
+
+"""
+    partial_pressure_vapor(param_set, ts::ThermodynamicState)
+
+The partial pressure of water vapor, given a thermodynamic state `ts`.
+"""
+@inline partial_pressure_vapor(
+    param_set::APS,
+    ts::ThermodynamicState{FT},
+) where {FT <: Real} = partial_pressure_vapor(
+    param_set,
+    air_pressure(param_set, ts),
+    PhasePartition(param_set, ts),
+)
+
+@inline partial_pressure_vapor(
+    param_set::APS,
+    ts::AbstractPhaseDry{FT},
+) where {FT <: Real} = FT(0)
+
+
+"""
+    vapor_pressure_deficit(param_set, T, p, q::PhasePartition)
+
+The vapor pressure deficit (saturation vapor pressure minus actual 
+vapor pressure, truncated to be non-negative) over liquid water for temperatures 
+above freezing and over ice for temperatures below freezing, given
+
+ - `param_set` an `AbstractParameterSet`, see the [`Thermodynamics`](@ref) for more details
+ - `T` air temperature
+ - `p` air pressure
+ - `q` [`PhasePartition`](@ref)
+"""
+@inline function vapor_pressure_deficit(
+    param_set::APS,
+    T::FT,
+    p::FT,
+    q::PhasePartition{FT},
+    Tᶠ = TP.T_freeze(param_set),
```

Comment:
Do we want users to be able to modify Tᶠ, or should we move Tᶠ inside the function?

Reply:
Users do not necessarily need to modify Tᶠ. However, my understanding is that this pattern increases efficiency in repeated function calls?

Reply:
Oh, I didn't know that. That's fine then.